librz/analysis/dwarf_process.c

```diff
@@ -857,42 +858,12 @@ static const char *map_dwarf_reg_to_x86_reg(ut64 reg_num, VariableLocationKind *
 /* https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi-1.9.html#DW-REG */
 static const char *map_dwarf_reg_to_ppc64_reg(ut64 reg_num, VariableLocationKind *kind) {
 	*kind = LOCATION_REGISTER;
-	switch (reg_num) {
-	case 0: return "r0";
-	case 1:
+	if (reg_num == 1) {
 		*kind = LOCATION_SP;
 		return "r1";
-	case 2: return "r2";
-	case 3: return "r3";
-	case 4: return "r4";
-	case 5: return "r5";
-	case 6: return "r6";
-	case 7: return "r7";
-	case 8: return "r8";
-	case 9: return "r8";
-	case 10: return "r9";
-	case 11: return "r10";
-	case 12: return "r11";
-	case 13: return "r12";
-	case 14: return "r13";
-	case 15: return "r15";
-	case 16: return "r16";
-	case 17: return "r17";
-	case 18: return "r18";
-	case 19: return "r19";
-	case 20: return "r20";
-	case 21: return "r21";
-	case 22: return "r22";
-	case 23: return "r23";
-	case 24: return "r24";
-	case 25: return "r25";
-	case 26: return "r26";
-	case 27: return "r27";
-	case 28: return "r28";
-	case 29: return "r29";
-	case 30: return "r30";
-	case 31: return "r31";
-	default:
+	} else if (reg_num < 32) {
+		return rz_str_newf("r%"PFMT64d, reg_num);
```

Comment:
this is wrong. It now allocates a string every time and the function returns `const char *` so I doubt it is freed by the caller.

Reply:
i agree. @XVilka can you open an issue to refactor this? because this needs more work.

Reply:
@ret2libc the simplification will involve changing more than just one function. Are you ok with merging this fix and do the rework of that file later?